promql/ast.go

```diff
@@ -115,6 +115,15 @@ type MatrixSelector struct {
 	series []storage.Series
 }
 
+// SubqueryExpr represents a subquery.
+type SubqueryExpr struct {
+	Expr       Expr
+	Range      time.Duration
+	Offset     time.Duration
+	Step       time.Duration
+	StepExists bool
```

Comment:
Don't feel strongly about this one, but I would have probably just used the zero value of `Step` to signal a non-specified step, unless there's a problem with that that I'm overlooking. It's pretty common to check if something is 0 and if yes, set it to some default.